registry/health/health_test.go

```diff
@@ -0,0 +1,63 @@
+package health
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+// TestReturns200IfThereAreNoChecks ensures that the result code of the health
+// endpoint is 200 if there are not currently registered checks.
+func TestReturns200IfThereAreNoChecks(t *testing.T) {
+	recorder := httptest.NewRecorder()
+
+	req, err := http.NewRequest("GET", "https://fakeurl.com/debug/health", nil)
+	if err != nil {
+		t.Errorf("Failed to create request.")
+	}
+
+	StatusHandler(recorder, req)
+
+	if recorder.Code != 200 {
+		t.Errorf("Did not get a 200.")
+	}
+}
+
+// TestReturns200IfThereAreOkOrWarningChecks ensures that the result code of the
+// health endpoint is 200 if there are health checks with the StatusWarning code
+func TestReturns200IfThereAreOkOrWarningChecks(t *testing.T) {
+	recorder := httptest.NewRecorder()
+
+	req, err := http.NewRequest("GET", "https://fakeurl.com/debug/health", nil)
+	if err != nil {
+		t.Errorf("Failed to create request.")
+	}
+
+	UpdateStatus(HealthStatus{"WarningStatus", StatusWarning})
+	UpdateStatus(HealthStatus{"OKStatus", StatusOK})
+	StatusHandler(recorder, req)
+
+	if recorder.Code != 200 {
+		t.Errorf("Did not get a 200.")
+	}
+}
+
+// TestReturns500IfThereAreErrorChecks ensures that the result code of the
+// health endpoint is 500 if there are health checks with the StatusError code
+func TestReturns500IfThereAreErrorChecks(t *testing.T) {
+	recorder := httptest.NewRecorder()
+
+	req, err := http.NewRequest("GET", "https://fakeurl.com/debug/health", nil)
```

Comment:
This testing approach mixes client-side `http.NewRequest` with an `http.Handler`. It is probably okay, but it would be better to use [net/http/httptest](http://golang.org/pkg/net/http/httptest/) with a test server. This ensure that request and response traverse http stack.

Let me know if you need an example.